components/director/internal/tenantfetcher/client.go

```diff
@@ -65,20 +70,59 @@ type Client struct {
 }
 
 // NewClient missing godoc
-func NewClient(oAuth2Config OAuth2Config, apiConfig APIConfig, timeout time.Duration) *Client {
+func NewClient(oAuth2Config OAuth2Config, authMode oauth.AuthMode, apiConfig APIConfig, timeout time.Duration) (*Client, error) {
```

Comment:
Can we add tests for the constructor that asserts the httpClient's configuration is what is expected based on the AuthMode?

Reply:
done